prog/dftb+/lib_io/formatout.F90

```diff
@@ -189,34 +188,38 @@ subroutine writeGenFormat_fid(fd, coord, species, speciesName, latVec, origin, t
         tPeriodic = .true.
       end if
       if (tFractional) then
-        write(fd, 100) nAtom, "F"
+        write(fd, 101) nAtom, "F"
       else if (tHelical) then
-        write(fd, 100) nAtom, "H"
+        write(fd, 101) nAtom, "H"
       else if (tPeriodic) then
-        write(fd, 100) nAtom, "S"
+        write(fd, 101) nAtom, "S"
       else
         call error("Unknown boundary conditions")
       end if
     else
-      write(fd, 100) nAtom, "C"
+      write(fd, 101) nAtom, "C"
     end if
 
-    write(formatSpecies, 101) nSpecies
-    write(fd, formatSpecies) (trim(speciesName(ii)), ii = 1, nSpecies)
+    do ii = 1, nSpecies
+      write(formatSpecies, "('(A',I0,')')")len(trim(speciesName(ii)))+1
+      write(fd, formatSpecies, advance='NO')trim(speciesName(ii))
+    end do
+    write(fd,*)
 
+    write(formatCoordinates, 102) floor(log10(real(nSpecies)))+1
```

Comment:
Please, don't mix `format` statements and format strings, this makes IO routines even harder to read.

Reply:
Point taken (was just sticking to the messy style of the existing routine).